pkg/dockerregistry/server/remoteblobgetter.go

```diff
@@ -324,8 +326,11 @@ func identifyCandidateRepositories(
 			}
 			ref = ref.DockerClientDefaults()
 			insecure := insecureByDefault
-			if tagRef, ok := is.Spec.Tags[tag]; ok {
-				insecure = insecureByDefault || tagRef.ImportPolicy.Insecure
+			for _, t := range is.Spec.Tags {
```

Comment:
external vs. internal... fixing external to map will break the backward compatibility.